GVFS/GVFS.Common/ProductUpgraderInfo.Shared.cs

```diff
@@ -11,14 +11,15 @@ public partial class ProductUpgraderInfo
         public const string LogDirectory = "UpgraderLogs";
         public const string DownloadDirectory = "Downloads";
         public const string HighestAvailableVersionFileName = "HighestAvailableVersion";
+        public const string GVFSUpgradeComponentName = UpgradeDirectoryName;
 
-        private const string RootDirectory = UpgradeDirectoryName;
-
-        public static bool IsLocalUpgradeAvailable(ITracer tracer)
+        public static bool IsLocalUpgradeAvailable(ITracer tracer, string gvfsDataRoot)
```

Comment:
Should these now be encapsulated into their own class? Maybe one that takes the `gvfsDataRoot` into the constructor?

Reply:
A couple things to consider.
1. This is a small static class (56 lines of code) has been created for use mostly in `GVFS.Hooks` project. `GVFS.Hooks` might be removed altogether in the future and the static `IsLocalUpgradeAvailable()` can then be moved back to `ProductUpgraderInfo.cs` which is a regular class.
2. There are other such `.Shared.cs` files, that are used by `GVFS.Hooks` (like  `GVFSEnlistment.Shared.cs`, `GVFSLock.Shared.cs`, `Paths.Shared.cs`, etc). The existing pattern in hooks project is to use `static` methods rather than instantiating objects for use.

We can discuss. If the goal is better test-ability, we can take this up as a refactoring effort along with the rest of the `static` classes used by `GVFS.Hooks`.